File: finstack-py/tests/golden/runners/calibration_hazard.py
"""Domain runner for credit hazard calibration golden fixtures."""

from __future__ import annotations

from tests.golden.runners._placeholders import reject_flattened_outputs
from tests.golden.schema import GoldenFixture


def run(fixture: GoldenFixture) -> dict[str, float]:
    """Require executable hazard calibration inputs before running."""
    return reject_flattened_outputs("hazard calibration runner", fixture)

File: finstack/valuations/tests/golden/runners/calibration_hazard.rs
//! Domain runner for credit hazard calibration golden fixtures.

use crate::golden::runner::DomainRunner;
use crate::golden::schema::GoldenFixture;
use std::collections::BTreeMap;

/// Credit hazard calibration golden runner.
pub struct CalibrationHazardRunner;

impl DomainRunner for CalibrationHazardRunner {
    fn run(&self, fixture: &GoldenFixture) -> Result<BTreeMap<String, f64>, String> {
        crate::golden::runners::reject_flattened_outputs("hazard calibration runner", fixture)
    }
}
